### PR TITLE
libcore: Implement cloned() for Option<&mut T>

### PR DESCRIPTION
**Pull request by panicbit**

### CODE REVIEW

**Comment by panicbit** [comment]
I'm a bit confused about how to deal with the stability attribute; I'd appreciate some input about that.

File: src/libcore/option.rs [hunk 1]
```diff
@@ -774,6 +774,25 @@ impl<'a, T: Clone> Option<&'a T> {
...
 }
 
+impl<'a, T: Clone> Option<&'a mut T> {
+    /// Maps an `Option<&mut T>` to an `Option<T>` by cloning the contents of the
+    /// option.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut x = 12;
```
Comment (oli-obk) [inline src/libcore/option.rs]: You need to add a `#![feature(option_ref_mut_cloned)]` to the doc example.

**Comment by arielb1** [comment]
I guess this does need an RFC + tracking issue + stabilization track.

**Comment by oli-obk** [comment]
> r? @oli-obk I think

I have no bors-powers.

> I guess this does need an RFC + tracking issue + stabilization track.

I don't think it needs an RFC, since we already have the implementation for `Option<&T>`, this is just a natural extension of that impl.

Definitely needs a tracking issue + must be unstable.

**Comment by aturon** [comment]
Hey, sorry for the late review!

cc @rust-lang/libs, how do y'all feel about adding this API?

**Comment by sfackler** [comment]
SGTM. There aren't any other methods on `Option<&T>` we'd also want to copy over, right?

**Comment by dtolnay** [comment]
LGTM as well. There are no other methods.

**Comment by alexcrichton** [comment]
👍

**Comment by carols10cents** [comment]
@aturon ping, it sounds like libs is cool with this?

**Comment by aturon** [comment]
@bors: r+

Thanks for the PR, and sorry for the delay!

**Comment by arielb1** [comment]
@bors retry

prioritizing rollup which includes this

**Pull request merged**